nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -59,6 +59,16 @@ let
       # Use the list of allowed interfaces if specified
       ${optionalString (allowInterfaces != null) "allowinterfaces ${toString allowInterfaces}"}
 
+      # Immediately fork to background if specified, otherwise wait for IP address to be assigned
+      ${{
+        background = "background";
+        any = "waitip";
+        IPv4 = "waitip 4";
+        IPv6 = "waitip 6";
+        both = "waitip 4\nwaitip 6";
+        default = "";
+      }.${cfg.wait}}
```

Comment:
The `default` here doesn't achieve anything. If you wanted a default you'd have to use `{ ... }.${cfg.wait} or ""`. But I don't think there's any reasonable default here anyways, so it should be okay to just not have one.

Reply:
Seems I was a bit tired. The dhcpcd default behaviour is selected with the option "if-carrier-up". And due to @gloaming's comment, this behaviour should differ from the other options available. The relevant part in the dhcpcd man page is:

> If any interface reports a working carrier then dhcpcd will try and obtain a lease before forking to the background, otherwise it will fork right away.